src/main/java/dbaccessor/user/UserAccessor.java

```diff
@@ -21,29 +22,48 @@
     @Autowired
     private UniqueIdGenerator uniqueIDGenerator;
 
-    /* Inserts new entry in User Table */
+    /**
+     * Generates a unique id for user and
+     * Inserts user in User Table
+     */
     public long insert(User user) {
         long id = uniqueIDGenerator.generateUniqueId("User");
         user.setUserId(id);
         spannerTemplate.insert(user);
         return id;
     }
 
-    /* Checks if a user with the given username or email-id already exists in User table */
-    public boolean checkIfUserExists(String username, String emailID) {
-        String SQLStatment = "SELECT Username FROM User WHERE Username=@Username OR EmailID=@EmailID";
+    /** Gets the UserID of the user having the given username and email-id */
+    public Optional<Long> getUserIdFromUsernameAndEmail(String username, String emailID) {
```

Comment:
Please use `OptionalLong` instead of `Optional<Long>`.
There are specialised classes for optional primitives like `OptionalInt`, `OptionalLong`, etc.

Reply:
Done.